src/server/scripts/EasternKingdoms/Karazhan/boss_prince_malchezaar.cpp

```diff
@@ -199,23 +190,57 @@ struct boss_malchezaar : public BossAI
             context.Repeat();
         }).Schedule(40s, [this](TaskContext context)
         {
-            Position pos =  me->GetRandomNearPosition(40.0);;
+            Talk(SAY_SUMMON);
 
-            if (Creature* RELAY = me->FindNearestCreature(NPC_RELAY, 100.0f))
+            me->GetCreaturesWithEntryInRange(relays, 1000.0f, NPC_INFERNAL_RELAY);
```

Comment:
grid calls are very expensive, consider limiting it to the boss area (this will pretty much search the entire instance)